packages/front-end/pages/metric/[mid].tsx

```diff
@@ -73,6 +73,7 @@ const MetricPage: FC = () => {
     getMetricById,
     metrics,
     segments,
+    project,
```

Comment:
We shouldn't use the project from definitions since it might not be the same as the metric's project.

Reply:
Makes sense. From what I can tell, in order to use the metric's projects, we'll need to update the `UserContext` so that it accepts an array of projects like we do with the backend `hasPermissions` and loop through each project to make sure the user has permission for every project in `metrics.projects`. This does resolve the edge case we discussed yesterday and removes the need for us to have specific error messaging in place should the user not have `runQueries` permission for every project a metric is a part of.

I do wonder if we should update the UX so the button is displayed, but disabled, with a contextual message in case any users are confused why they can't refresh the metric, but I think that should be done in a separate PR.